### ISSUE TITLE
Parser bug while typing Vector constructor referencing another vector component

### ISSUE THREAD

**Issue by lostminds** [godotengine/godot#68459]
### Godot version

4.0.beta4

### System information

macOS 12.6.1

### Issue description

While harmless I've run into a parser bug error I can reproduce in the code editor: Every time I type a specific code pattern I get this error in the console:
"Parser bug (please report): Trying to check compatibility of unset value type"
It happens as I type and doesn't block anything

### Steps to reproduce

Basically it seems to affect all Vector constructors (like Vector2, Vector3, Vector2i etc) and happens if you start typing a constructor, referencing in the first component a component of another Vector. 
For example:
```
var vectorA:Vector2
var vectorB:Vector2
vectorB = Vector2(vectorA.x
```
Note that the last row is unfinished, so that will give you another error if you don't finish it. The point is that it's at this point in typing the Parser error is triggered

### Minimal reproduction project

- Open a new script file
- in the _ready function type the following:
```
var vectorA:Vector2
var vectorB:Vector2
vectorB = Vector2(vectorA.x
```
I get the same error using other vector types as well, and you can probably get them for other cases as well

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/issues/44375.

**Comment by lostminds** [comment]
Tested and I'm no longer getting this parser error in 4.0 beta 10. The related issue is also closed so I'm assuming this is fixed and I'm closing this issue.

**Comment by lostminds** [comment]
It seems I was unfortunately a bit hasty closing this, after testing again I was still able to get the same error as described above in the Output log.

**Comment by lostminds** [comment]
Looks like this is fixed in 4.0.stable, closing this
